### PR TITLE
Issue #3955: corrected tokens so all are required

### PR DESCRIPTION
**Pull request by rnveach**
Issue #3955

it is breaking compatibility .... will need review from all.

### CODE REVIEW

**Comment by romani** [comment]
it is breaking compatibility .... will nee review from all.

**Review by pbludov — APPROVED** [review]

**Comment by nick-mancuso** [comment]
Github, generate site

**Review by nick-mancuso — APPROVED** [review]
This is a big improvement to this check; documentation and usage is clear now.

**Review by strkkk — APPROVED** [review]

**Comment by rnveach** [comment]
@romani ping

**Comment by romani** [comment]
Github, generate web site

**Review by romani — CHANGES_REQUESTED** [review]
Do we have test that shows that user can define FQN of inner class to be validated ? 

items:

**Comment by rnveach** [comment]
> Do we have test that shows that user can define FQN of inner class to be validated ?

InputIllegalInstantiationSemantic2 has:
````
classes = java.lang.Boolean,com.puppycrawl.tools.checkstyle.checks.coding.illegalinstantiation.\
          InputModifier,java.io.File,java.awt.Color
````

Please note `InputModifier`.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/IllegalInstantiationCheck.java [hunk 1]
```diff
@@ -95,12 +87,7 @@
  * </pre>
  * <p>
  * To configure the check to find instantiations of {@code java.lang.Boolean}
- * and {@code java.lang.Integer}. NOTE: Even if property {@code tokens}
- * is completely removed from the following configuration, Checkstyle will produce
- * the same results for violation. This is because if property {@code tokens} is not
- * defined in the configuration, Checkstyle will supply it with list of default tokens
- * {@code CLASS_DEF, LITERAL_NEW, PACKAGE_DEF, IMPORT} for this check. The property is
- * defined in this example only to provide clarity:
+ * and {@code java.lang.Integer}:
```
[the remark this answers is not in the document]
Comment (rnveach) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/IllegalInstantiationCheck.java:90]: Done.

**Comment by romani** [comment]
Github, generate web site

**Review by romani — APPROVED** [review]
Ok to merge

**Pull request merged**